### PR TITLE
Lift SQLite foreign key additions into create table statement

### LINKED ISSUE #2590: SQLite Migrations: Lift foreign keys to create table operation
The model differ breaks foreign key cycles in create table operations by pulling one out into a separate add foreign key operation. Unfortunately, SQLite can't add foreign keys to a table without rebuilding it. The SQLite migration SQL generator should push them back into their create table operation since cycles don't matter when foreign keys aren't enforced.

### PR DESCRIPTION
**Pull request by natemcmaster**
Fix #2590 

/cc: @bricelam

### CODE REVIEW

File: test/EntityFramework.Sqlite.FunctionalTests/SqliteDatabaseCreationTest.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+using Microsoft.Data.Entity.Infrastructure;
+using Microsoft.Data.Entity.Sqlite.FunctionalTests;
+using Microsoft.Data.Entity.Storage;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging;
+using Xunit;
+
+namespace Microsoft.Data.Entity.Sqlite
+{
+    public class SqliteDatabaseCreationTest : IClassFixture<SqliteDatabaseCreationFixture>
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.FunctionalTests/SqliteDatabaseCreationTest.cs]: No need for a functional test, IMHO. I would just assert the SQL with a new unit test in [SqliteMigrationSqlGeneratorTest](https://github.com/aspnet/EntityFramework/blob/dev/test/EntityFramework.Sqlite.Tests/Migrations/SqliteMigrationSqlGeneratorTest.cs).

**Comment by bricelam** [comment]
:shipit:

File: test/EntityFramework.Sqlite.FunctionalTests/SqliteDatabaseCreationTest.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        public void It_lifts_foreign_key_creation_to_create_table_operation()
+        {
+            using (var context = _fixture.CreateContext())
+            {
+                var creator = context.GetService<IDatabaseCreator>() as RelationalDatabaseCreator;
+                creator.CreateTables();
+            }
+
+            Assert.StartsWith("CREATE TABLE ", TestSqlLoggerFactory.Sql);
+            Assert.Equal(2, TestSqlLoggerFactory.SqlStatements.Count);
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.FunctionalTests/SqliteDatabaseCreationTest.cs]: This also doesn't assert that the foreign key was actually added...

Comment (natemcmaster) [inline test/EntityFramework.Sqlite.FunctionalTests/SqliteDatabaseCreationTest.cs]: Good point. This test used the example from #3041. Will move to a simpler test as you suggested.

File: test/EntityFramework.Sqlite.Tests/Migrations/SqliteMigrationSqlGeneratorTest.cs [hunk 3]
```diff
@@ -19,6 +19,35 @@ protected override IMigrationsSqlGenerator SqlGenerator
...
+                                {
+                                    ClrType = typeof(int),
+                                    Name = "FlavorId",
+                                    ColumnType = "INT"
+                                }
+                        }
+                }, new AddForeignKeyOperation
+                    {
+                        Table = "Pie",
+                        PrincipalTable = "Flavor",
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.Tests/Migrations/SqliteMigrationSqlGeneratorTest.cs]: Self-refs are actually not broken out into their own operations. :smile:

**Pull request merged**